component/storageutil/formattedstore/formattedstore.go

```diff
@@ -76,9 +83,6 @@ func (f *FormattedProvider) OpenStore(name string) (spi.Store, error) {
 
 	name = strings.ToLower(name)
 
-	f.lock.Lock()
```

Comment:
Fixed. In order to avoid a deadlock situation With SetStoreConfig I had to refactor some of the OpenStore code into a separate method. Can you also take a look at SetStoreConfig and let me know if the locks there look ok?